workspace_api/views.py

```
@@ -98,7 +99,20 @@ async def create_workspace(
         )
     )
 
-    create_bucket(workspace_name=workspace_name)
+    # create_bucket(workspace_name=workspace_name)
+    # TODO: specify post body
+    response = requests.post(bucket_endpoint_url, data={})
+    if response.status == 200:
+        # we have a bucket created, we create the secret and continue setting up workspace
+        create_bucket_secret(workspace_name=workspace_name, credentials=response.body)
+
+    elif response.status == 201:
+        # async approach
+        # wait for the secret to be created
+        # TODO determine the timeout & time interval values from config ?
+        background_tasks.add_task(wait_for_secret, response.body, 100, 10)
+
```

If status is in 400/500 range we need to propagate the error in the response

---

https://fastapi.tiangolo.com/tutorial/handling-errors/